packages/synapse-sdk/src/storage/manager.ts

```diff
@@ -133,14 +137,32 @@ export class StorageManager {
       }
     }
 
-    // Get the context to use
-    const context = options?.context ?? (await this.createContext(options))
+    if (options?.contexts != null && options.contexts.length > 0) {
+      if (options?.context != null) {
+        throw createError('StorageManager', 'upload', "Cannot specify both 'context' and 'contexts'")
+      }
+    }
 
-    // Upload using the context with piece metadata
-    return await context.upload(data, {
-      ...options?.callbacks,
-      metadata: options?.metadata,
-    })
+    // Get the context to use
+    const contexts = options?.contexts ?? [options?.context ?? (await this.createContext(options))]
+    // Convert once
+    const dataBytes = data instanceof ArrayBuffer ? new Uint8Array(data) : data
+    // Calculate pieceCid once
+    const pieceCid = Piece.calculate(dataBytes)
+
+    // Upload using the contexts with piece metadata
+    return await Promise.all(
```

Comment:
If one of 5 contexts fails, this will reject all of them.. 

Will those background uploads be aborted? Currently doesn't look like it (no `signal` support)
Should we do `allSettled` and have a return that includes successes/failures?